h2o-core/src/main/java/water/rapids/FeatureImportance4BBM.java

```diff
@@ -0,0 +1,129 @@
+
+    public FeatureImportance4BBM(Model model, Frame data_set, Frame prediction_frame) {
+        _model = model;
+        _train_fr = data_set;
+        _data_in_holdOut_form = false;
+        _validate_dataset = null;
+        _predcited_frame = prediction_frame;
+        _responce_col = _train_fr.name(0); // first column as default
+    }
+
+    public FeatureImportance4BBM(Model model, Frame data_set, Frame prediction_frame, String response_col) {
+        _model = model;
+        _train_fr = data_set;
+        _data_in_holdOut_form = false;
+        _validate_dataset = null;
+        _predcited_frame = prediction_frame;
+        _responce_col = response_col;
+    }
+    
+    
+    //MSE set as default loss function
+    public void getFeatureImportance(){
+        getFeatureImportance("MSE");
+    }
+    
+    public void getFeatureImportance(String metric) {
+        if (!_data_in_holdOut_form) {
+            //TODO split the data if needed
+        }
+
+        String[] features = _train_fr.names();
+        HashMap<String, Double> FI = new HashMap<String, Double>(_train_fr.vecs().length);
+
+        // original error calc using MSE or AUC as a loss function
+        if (metric.equals("MSE")){
+            sq_err = new MathUtils.SquareError().doAll(_train_fr.vecs()[1], _predcited_frame.vecs()[0])._sum;
+            original_mse = sq_err / _predcited_frame.numRows();
+        }   else    {
+            og_auc = new AUC2(_train_fr.vec(_responce_col), _predcited_frame.vecs()[0]); //claisifiication metric
+        }
+        for (int f = 1; f < _train_fr.numCols(); f++) //f = 0 is predicted col
+        {
+            //permuate vector
+            Vec shuffled_feature = ShuffleVec.ShuffleTask.shuffle(_train_fr.vec(features[f]));
+            Vec og_feature = _train_fr.replace(f, shuffled_feature);
+            DKV.put(_train_fr); // "Caller must perform global update (DKV.put) on this updated frame"
+
+            // score the model again and compute diff
+            Frame new_score = _model.score(_train_fr);
+
+            //calucalate error from new score prediction
+            if (metric.equals("MSE")){ // regression
+                sq_err = new MathUtils.SquareError().doAll(_train_fr.vec(_responce_col), new_score.vecs()[0])._sum;
+                FI.put(features[f], original_mse / (sq_err/new_score.numRows())); // Hashmap of HI feature->FI_mse
+            }   else    {  //clasification
+                AUC2 auc = new AUC2(_train_fr.vec(_responce_col), new_score.vecs()[0]); //claisifiication metric
+                FI.put(features[f], og_auc._auc - auc._auc);
+            }
+            
+            //return the original data
+            _train_fr.replace(f, og_feature);
+            DKV.put(_train_fr); // "Caller must perform global update (DKV.put) on this updated frame"
+
+            new_score.remove(); // clearing leaks
+        }
+        /*return */HashMap<String, Double> sorted_FI_mse = sortHashMapByValues(FI);
```

Comment:
Why are you use HashMap here? There should be similar implementation on how to sort feature importance output for example in tree algorithms. Try to find it.